### PR TITLE
CPO-583 Reduce output from Terraform

### PR DESCRIPTION
**Pull request by tom-mckay**
Add the procore_terraform_wrapper.py script to reduce the output from Terraform and fix an issue with return codes not passing through tfmask.

https://procoretech.atlassian.net/browse/CPO-583

### CODE REVIEW

**Review by rarkebauer — CHANGES_REQUESTED** [review]
I don't understand the README.bak.md and blank README.md. Did you mean to leave the README.md as is in this PR?

Can you also include a short readme in the procore_terraform_wrapper with the commands to run the script and the tests?

**Comment by tom-mckay** [comment]
> I don't understand the README.bak.md and blank README.md. Did you mean to leave the README.md as is in this PR?
> 
> Can you also include a short readme in the procore_terraform_wrapper with the commands to run the script and the tests?

I moved the upstream's README.md to README.bak.md and replaced README.md with content specific to our fork of the upstream. It's not blank.

**Comment by rarkebauer** [comment]
whoops I didn't realize github was hiding it. Can you still add a note about running/testing the procore_terraform_wrapper locally?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (brandonpollard) [inline .vscode/settings.json:1]: Shouldn't this be ignored?

Comment (tom-mckay) [reply .vscode/settings.json:1]: Not necessarily. This file is helpful for vscode users as it sets up the linters to run automatically.

File: README.md [hunk 2]
```diff
@@ -1,169 +1,129 @@
...
 
-1. Run the tests:
-    ```bash
-    helm test my-atlantis
-    ```
+## Running tests
+Run the tests after any changes to the python code.
+
+```
+pip3 install -f requirements.txt
```
Comment (rarkebauer) [inline README.md]:
```suggestion
pip3 install requirements.txt
```

File: README.md [hunk 3]
```diff
@@ -1,169 +1,129 @@
...
----
-repos:
-- id: /.*/
-  apply_requirements: [approved, mergeable]
-  allowed_overrides: [apply_requirements, workflow]
-  allow_custom_workflows: true
-
-or use --repo-config-json='{"repos":[{"id":"/.*/", "apply_requirements":["approved", "mergeable"], "allowed_overrides":["apply_requirements","workflow"], "allow_custom_workflows":true}]}'
 ```
+usage: procore_terraform_wrapper.py [-h] --action {init,plan,apply}
```
Comment (rarkebauer) [inline README.md]:
```suggestion
usage: ./procore_terraform_wrapper.py [-h] --action {init,plan,apply}
```

Comment (tom-mckay) [reply README.md]: I generated this by copy-pasting the output of `./procore_terraform_wrapper.py -h`. The name of the script is automatically generated by argparse. I'd rather not change it manually in the readme.

File: README.md (hunk 2 above)
Comment (tom-mckay) [reply README.md]: This was supposed to be `-r`. Fixed.

File: README.md (hunk 3 above)
Comment (rarkebauer) [reply README.md]: I would prefer to make the readme the source of copy/pasteable commands. I think it is easier when someone is completely unfamiliar with the repo. But that doesn't need to be a blocker if we disagree

**Review by rarkebauer — APPROVED** [review]

**Comment by tom-mckay** [comment]
I'm going to hold off on merging this until we have the staging atlantis bot and repo ready.

**Comment by tom-mckay** [comment]
Ok, this is ready for re-review. I fixed issues that came up during testing:
- Use arguments instead of environment variables to make it easier to test.
- Use `shell=True` to fix strange issues when run via Atlantis

A PR is available to test out the new code here: procore/atlantis-test-repository#3

**Review by rarkebauer — APPROVED** [review]

**Pull request merged**